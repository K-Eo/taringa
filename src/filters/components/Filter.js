import React from 'react'
import { NavLink } from 'react-router-dom'

const Filter = ({ displayName, exact, pathname }) => {
  return (
    <NavLink
      activeClassName="active"
      className="nav-item nav-link"
      exact={exact}
      to={{
        pathname: pathname,
        state: { clear: true },
      }}
    >
      {displayName}
    </NavLink>
  )
}

Filter.defaultProps = {
  displayName: 'Nav Link',
  exact: false,
  pathname: '/navlink',
}

export default Filter
