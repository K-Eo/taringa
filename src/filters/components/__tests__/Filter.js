import React from 'react'

import Filter from '../Filter'

describe('Filter', () => {
  it('exists', () => {
    expect(Filter).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Filter />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with props', () => {
    const wrapper = shallow(<Filter displayName="foo" exact pathname="/bar" />)

    expect(wrapper).toMatchSnapshot()
  })
})
