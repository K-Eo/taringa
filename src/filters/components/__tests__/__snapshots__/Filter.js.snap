// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders default 1`] = `
<NavLink
  activeClassName="active"
  ariaCurrent="true"
  className="nav-item nav-link"
  exact={false}
  to={
    Object {
      "pathname": "/navlink",
      "state": Object {
        "clear": true,
      },
    }
  }
>
  Nav Link
</NavLink>
`;

exports[`Filter renders with props 1`] = `
<NavLink
  activeClassName="active"
  ariaCurrent="true"
  className="nav-item nav-link"
  exact={true}
  to={
    Object {
      "pathname": "/bar",
      "state": Object {
        "clear": true,
      },
    }
  }
>
  foo
</NavLink>
`;
